### DISCUSSION TITLE
Tips for fast search results?

### DISCUSSION THREAD

**Discussion by ecker00** [searxng/searxng#1738] [General]
I'm wondering what are some of the best practices when it comes to improving performance and response time for SearXNG? I've not seen much discussion on it, and there are a lot of parameters to tweak. How are the best servers achieving sub-second response times?

Just to get a sense of how fast different sites where, I did some very rough testing estimates in my browser.

DOMContentLoaded timing in Firefox:
- Quant ~250-500 ms
- DuckDuckGo ~550-750 ms
- Google and Bing ~900-1300 ms
- [Tiekoetter's Searxng](https://searx.tiekoetter.com/) instance ~950-1300 ms (Quite good!)
- My own Searxng instance ~6200-6600 ms  (On fast fiber, M1 mac server)

My own SearXNG server usually takes more than 6 seconds to process a basic request, any tips?

**Comment by tiekoetter** [comment]
Usually having only the upstream engines enabled which you really need / want to use helps a lot. If that doesn't help you can look on the engine stats page to see which engines are slow and disable them.

This should give you some insight in what might be slowing you down.

**Reply by ecker00** [reply]
Thank you for some pointers. 👍 If other people have suggestion, be nice to post them here as a future reference.
